collectors/scm/github-graphql/src/main/java/com/capitalone/dashboard/collector/DefaultGitHubClient.java

```diff
@@ -1042,6 +1056,42 @@ private static DateTime getDate(DateTime dateInstance, int offsetDays, int offse
         return dateInstance.minusDays(offsetDays).minusMinutes(offsetMinutes);
     }
 
+
+    private JSONObject getDataFromRestCallPost(String url, GitHubRepo repo, String password, String personalAccessToken, JSONObject query) throws MalformedURLException, HygieiaException {
+        ResponseEntity<String> response = makeRestCallPost(url, repo.getUserId(), password, personalAccessToken, query);
+        JSONObject data = (JSONObject) parseAsObject(response).get("data");
+        JSONArray errors = getArray(parseAsObject(response), "errors");
+
+        if (!CollectionUtils.isEmpty(errors)) {
```

Comment:
My only comment here is that this file is riddled with deeply nested ifs and loops. Could you update this to fail fast to remove some of nests?